### DISCUSSION TITLE
support INITIAL_CONTACT

### DISCUSSION THREAD

**Discussion by XYQ202** [strongswan/strongswan#1135] [Q&A]
Hello , Team
      Do we support INITIAL_CONTACT - this message is sent in IKE header by client , it will tell peer end (SecGW) that this is my first session and clean the stale connections. Useful in case of reestablishing connections after outage.

I didn't find strongswan's configuration for INITIAL_CONTACT, as follows:
https://docs.pulsesecure.net/WebHelp/PCS/9.1R1/AG/Content/PCS/PCS_AdminGuide_9.1R1/Enabling_the_IKEv2_Initial_Contact.htm#:~:text=To%20enable%20the%20IKEv2%20access,process%20INITIAL_CONTACT%20request%20check%20box. This will be critical for production IPSEC deployments.

**Comment by strongX509** [comment]
Yes, **INITIAL_CONTACT** is supported by strongSwan and sent by first-time IKEv2 initiators by default. For details on the mechanism, see the **unique** parameter in [swanctl.conf](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html#_connections).
